src/uct/ib/rc/verbs/rc_verbs_ep.c

```diff
@@ -385,9 +385,26 @@ ucs_status_t uct_rc_verbs_ep_flush(uct_ep_h tl_ep, unsigned flags,
 
 ucs_status_t uct_rc_verbs_ep_fence(uct_ep_h tl_ep, unsigned flags)
 {
-    uct_rc_verbs_ep_t *ep = ucs_derived_of(tl_ep, uct_rc_verbs_ep_t);
+    uct_rc_verbs_iface_t *iface = ucs_derived_of(tl_ep->iface, uct_rc_verbs_iface_t);
+    uct_rc_verbs_ep_t *ep       = ucs_derived_of(tl_ep, uct_rc_verbs_ep_t);
+    ucs_status_t status;
+
+    UCT_TL_EP_STAT_FENCE(ucs_derived_of(tl_ep, uct_base_ep_t));
+
+    if (iface->super.config.fence_mode == UCT_RC_FENCE_MODE_NONE) {
+        return UCS_OK;
+    }
 
-    return uct_rc_ep_fence(tl_ep, &ep->fi, 1);
+    status = uct_rc_ep_flush(&ep->super, iface->config.tx_max_wr, flags);
```

Comment:
why need a flush?

Reply:
flush is used to identify existing resources, ```uct_rc_ep_flush``` doesn't make flush - it just checks for existing ongoing ops and resources